### ISSUE TITLE
[Bug] Hexadecimal Unicode Input Results in Incorrect Character

### ISSUE THREAD

**Issue by NotRexButCaesar** [pit-ray/win-vind#187]
# About Bug

Hexadecimal Unicode input is altered (seemingly randomly—I see no pattern) changed when the program is running

# Steps to reproduce
1. Enable Hexadecimal Unicode input
2. Activate Win Vind
3. enable insert mode
4. focus a text field
5. type "+301" on the numberpad with alt held
6. the character - is created, rather that the expected ́

(https://unicodelookup.com/#0x301/1)

# Environments

|Environments||
|:---:|:---|
|Windows Edition|10 Home|
|Windows Version|[e.g. 10.0.19045]|
|Windows Build|[e.g. 19045]|
|Architecture|64bit|
|win-vind Version|e.g. 5.2.0|
|win-vind Install Type|zip|

**Comment by pit-ray** [comment]
Insert mode should be outside the jurisdiction of win-vind.
Isn't that a bug in your text editor?

**Comment by NotRexButCaesar** [comment]
I do not think so: It works fine when I close win-vind using task manager, and the incorrect input persists across multiple different programs, from text editor to web browser.On 5 Aug 2023, at 10:17, kaz ***@***.***> wrote:﻿
Insert mode should be outside the jurisdiction of win-vind.
Isn't that a bug in your text editor?

**Comment by pit-ray** [comment]
When I set the keyboard layout to EN(US), I could confirm the problem that HexNumpad does not work only when win-vind is running. However, I could not reproduce the behavior where `-` is typed instead. It may depend on the environment.

I put some debugging code in the keyboard hooks and conducted various tests, but could not find the cause of the problem.

Please let me know if you can find out the cause of the problem.

**Comment by pit-ray** [comment]
I added this problem into the project board.
https://github.com/users/pit-ray/projects/2?pane=issue&itemId=35151380

**Comment by NotRexButCaesar** [comment]
I noticed that using the arrow keys sometimes affects the behavior: if I have a macro which inputs Unicode, using the arrow keys will sometimes allow the macro to input the correct character if it is used immediately after the arrow keys.

It may be this information is helpful.

**Comment by pit-ray** [comment]
Thanks you for investigating this problem.

Which macro you mentioned, the one in win-vind or the one in the other tool?
If it is a win-vind, please give me a sample .vindrc for reference.

I will look into the causality between arrow keys and HexNumpad.
Thanks.

**Comment by NotRexButCaesar** [comment]
It is an external macro in my keyboard, so it should be unrelated to any software: it is just fast keyboard input.  This makes me believe it probably has something to do with a timer in win-vind, or maybe a timer in Windows.

**Comment by pit-ray** [comment]
As seen in #158 and #153, the relation between the timer of the keyboard and that of Windows may be a factor in many issues.

However, if win-vind supports the configurable timer for updates, we must consider a lot of aspects, such as the mouse speed or the CPU usage and so on.

I think that we need to investigate the direct factor of this issue.
Thanks for the important information.
